minetest/minetest
# Clean up l_object.cpp

So, I've decided cleaning up MT code on the go. I think a consistent ordered code is vital both for core devs and for possible contributors. This is, in short, what I did:

* consistent names
  * `ObjectRef = obj`
  * `ServerActiveObject = sao`
  * `LuaEntitySAO = entitysao`
  * `PlayerSAO = playersao`
  * `RemotePlayer = player`
* consistent structure: check for object, variables declaration, execution+return. These sections are separated by an empty line. When functions were too long, the execution got split up accordingly.
* removed all `NULL` checks. Now there are `nullptr`s instead, as I find it more straightforward than `!something`
* removed useless comments, e.g. `//Do it`
* removed `is_player_connected`, a deprecated method that wasn't even listed in the API
* descriptions above functions now match the API
* removed info about parameters in the functions description: they're already explained in the API
* edited the API where needed
* simplified a few methods to make them more readable e.g. `set_bone_position`
* added missing Lua checks ( partially solves #1794 )

Things I'm not sure about, core dev help needed:
* ~how to name `p` in `set_sprite(...)`~
* the old `-1` value of HUD IDs, returned when ID was not a number. I had a look at the code and it doesn't look to me was meaningful at all so I simply removed it, but a double check is always better
* ~debug lines at 2249 and 2257. Do you want to keep them?~
* TODO at 483: has it been solved?

## To do

This PR is Ready for Review.

## How to test

Well: check the code. Also, I might have made a few mistakes as I'm human, even if I double-checked my edits twice. So idk, take a solid game, play it a little bit and see if it all works accordingly.

Lastly, I want to underline how, imho, this should be high priority: I've seen things which were pretty messed up, like edits pushed a couple of weeks ago going completely against the guidelines (the minimap), or useless 8 year old comments who nobody apparently decided to remove. A tidy environment helps a lot. And I've actually had fun doing it, it was relaxing
